Adds MessageSend component

webinista opened the pull request:
Work in progress on department-of-veterans-affairs/messaging-fe#31
- Eliminated most set\* methods in favor of a single setMessageField function.
- Adds placeholder method for fetching sender's name
- Adds placeholder confirm delete draft method

webinista committed:
Fixing PropTypes bug in ButtonDelete.

webinista committed:
Styling MessageSend buttons.

webinista committed:
Adding MessageSend component to Compose.

webinista committed:
Adding onSave, onSend props to MessageSend.

webinista committed:
Adding sendMessage, saveMessage actions. Refactored initialState for compose to better separate configs for that page vs. message object.

webinista committed:
Merging latest changes from master.

webinista committed:
Adds MessageSend component

Work in progress on department-of-veterans-affairs/messaging-fe#31

- Eliminated most set* methods in favor of a single setMessageField function.
- Adds placeholder method for fetching sender's name
- Adds placeholder confirm delete draft method

webinista committed:
Removing extra selector

webinista committed:
Making sure SHOW_CONFIRM_DELETE is defined.

webinista committed:
Merge branch 'department-of-veterans-affairs/messaging-fe#31' of github.com:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/messaging-fe#31

U-DON commented on src/js/messaging/reducers/compose.js:
```diff
@@ -1,18 +1,39 @@
+    attachments: []
+  },
+  modals: {
+    deleteconfirm: {
```
`deleteConfirm`

U-DON commented on src/js/messaging/reducers/compose.js:
```diff
@@ -32,17 +53,19 @@ function getRecipients(recipients) {
+    case FETCH_SENDER_SUCCESS:
+      return set('message.sender', action.sender, state);
+    case SHOW_CONFIRM_DELETE:
+      return set('modals.deleteconfirm.visible', !state.modals.deleteconfirm.visible, state);
```
If the one action switches the dialog on and off, I would call it something like `TOGGLE_CONFIRM_DELETE` rather than just `SHOW_*`.

U-DON commented on src/js/messaging/containers/Compose.jsx:
```diff
@@ -1,64 +1,79 @@
 
   render() {
+    const message = this.props.compose.message;
+    const recipients = this.props.compose.recipients;
```
Specifically pull these out into `Compose`'s props in `mapStateToProps`.

U-DON commented on src/js/messaging/actions/compose.js:
```diff
@@ -1,22 +1,27 @@
+export function setMessageField(path, field) {
   return {
-    type: SET_SUBJECT,
+    type: SET_MESSAGE,
```
`SET_MESSAGE_FIELD`

U-DON commented on src/js/messaging/containers/Compose.jsx:
```diff
@@ -1,64 +1,79 @@
 
   handleRecipientChange(valueObj) {
-    this.props.setRecipient(valueObj);
+    this.props.setMessageField('message.recipient', valueObj);
```
I think I preferred the old way, even though it creates a larger set of names to keep track of. This seems to be too closely coupled with the usage of the specific `set` function that updates the store, as the meaning of it isn't fully clear unless you follow the stack to that call and know that's how it works.

webinista replied:
I think that's a small risk.

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#31

webinista committed:
Changing constant names.

- Changing SHOW_CONFIRM_DELETE to TOGGLE_CONFIRM_DELETE
- Changing SET_MESSAGE to SET_MESSAGE_FIELD

webinista committed:
Mapping state to props idiomatically.

U-DON commented on src/js/messaging/reducers/compose.js:
```diff
@@ -32,17 +53,19 @@ function getRecipients(recipients) {
+    case FETCH_SENDER_SUCCESS:
+      return set('message.sender', action.sender, state);
+    case TOGGLE_CONFIRM_DELETE:
+      return set('modals.deleteconfirm.visible', !state.modals.deleteconfirm.visible, state);
```
Update to `deleteConfirm` here too.

webinista committed:
Camelcasing deleteConfirm everywhere.
